02-express-tutorial/answers/01-http-basics.js

```diff
@@ -0,0 +1,21 @@
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  const url = req.url;
+
+  if (url === '/') {
+    res.writeHead(200, { 'content-type': 'text/html' });
+    res.write('<h1>Home Page</h1>');
+    res.end();
+  } else if (url === '/about') {
+    res.writeHead(200, { 'content-type': 'text/html' });
+    res.write('<h1>About Page</h1>');
+    res.end();
+  } else {
+    res.writeHead(404, { 'content-type': 'text/html' });
+    res.write('<h1>Page Not Found</h1>');
+    res.end();
+  }
+});
```

Comment:
What are the biggest downsides of this built-in node HTTP server compared to express?

Reply:
Hi 

> Great work!
Thank you, Jack.

Reply:
> What are the biggest downsides of this built-in node HTTP server compared to express?

There are some limitations in node compared to express. As an example:  the built-in node HTTP server doesn't provide any built-in support for middleware and routing or boilerplate code, unlike express.

Reply:
Middleware & routing -- definitely a good answer. Not sure what you mean by boilerplate? Maybe the idea that you need to write a bit more boilerplate to use the raw node http library? That would make sense. Some libraries have CLI tools that will help you generate boilerplate. This is especially common in the Java world. In JS, there's more of a philosophy of designing things such that they don't need to have boilerplate in the first place.

Reply:
Yes, like using the raw Node.js HTTP may require more effort and time to set up compared to using a web framework like Express.

Reply:
Makes sense! Nice work!